packages/smooth_app/lib/pages/scan/scanner_overlay.dart

```diff
@@ -0,0 +1,114 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:smooth_app/data_models/continuous_scan_model.dart';
+import 'package:smooth_app/pages/scan/scan_page_helper.dart';
+import 'package:smooth_app/widgets/smooth_product_carousel.dart';
+import 'package:smooth_ui_library/animations/smooth_reveal_animation.dart';
+import 'package:smooth_ui_library/widgets/smooth_view_finder.dart';
+import 'package:visibility_detector/visibility_detector.dart';
+
+class ScannerOverlay extends StatelessWidget {
+  const ScannerOverlay({
+    required this.scannerWidget,
+    required this.model,
+    required this.restartCamera,
+    required this.stopCamera,
+  });
+
+  final Widget scannerWidget;
+  final ContinuousScanModel model;
+  final Function() restartCamera;
+  final Function() stopCamera;
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (
+        BuildContext context,
+        BoxConstraints constraints,
+      ) {
+        final Size screenSize = MediaQuery.of(context).size;
+        final Size scannerSize = Size(
+          screenSize.width * 0.6,
+          screenSize.width * 0.33,
+        );
+        final double carouselHeight =
+            constraints.maxHeight / 1.81; // roughly 55% of the available height
+        final double buttonRowHeight = areButtonsRendered(model) ? 48 : 0;
+        final double availableScanHeight =
+            constraints.maxHeight - carouselHeight - buttonRowHeight;
+        // Padding for the qr code scanner. This ensures the scanner has equal spacing between buttons and carousel.
+        final EdgeInsets qrScannerPadding = EdgeInsets.only(
+            top: (availableScanHeight - scannerSize.height) / 2 +
+                buttonRowHeight);
+
+        return VisibilityDetector(
+          key: const ValueKey<String>('VisibilityDetector'),
+          onVisibilityChanged: (VisibilityInfo info) {
+            if (info.visibleFraction == 0.0) {
+              stopCamera.call();
+            } else {
+              restartCamera.call();
+            }
+          },
+          child: Stack(
+            children: <Widget>[
+              Container(
```

Comment:
The question I have is why do we need this container ?

Reply:
This container is helpful for a kind of scanner starting animation, the camera preview takes a little time to show up, and until then just a black screen with a scanner logo is shown until the preview animated before it. It's not from me but I'd leave it there.

Reply:
Got it, thanks, that's very usedul information, could you please document it